include/lp_hash_table.h

```diff
@@ -0,0 +1,86 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <type_traits>
+
+template <typename T> struct lp_hash_table {
+  static_assert(std::is_pointer<T>::value, "T must be a pointer");
```

Comment:
Ok, this difficult to think about but check the [is_pointer doc](https://en.cppreference.com/w/cpp/types/is_pointer).
It says: Checks whether T is a pointer to object or _a pointer to function_. We don't want to handle the latter. It's not of great importance to check that but if you want to think about it, you may do.

Reply:
So `!std::is_pointer<std::function<T> *>::value` should work right?

Reply:
Actually for directing you in this path and making you waste time. This is C++ magic that we should not be focusing on.
One workaround is this:
```
#include <functional>

template <typename T> struct A {
  static const bool is_pointer = std::is_pointer<T>::value;
  static const bool is_function_pointer =
      is_pointer
          ? std::is_function<typename std::remove_pointer<T>::type>::value
          : false;
  static_assert(is_pointer && !is_function_pointer,
                "Template parameter should be of pointer type (but not a "
                "function pointer).");
};

A<int *> a;  // Pass
A<void (*)()> b;  // Fail
```
You may ignore that.

Reply:
It's ok the missing piece was `std::remove_pointer`.